### ISSUE TITLE
Use asynchronous aiohttp for making requests

### ISSUE THREAD

**Issue by rominf** [pycontribs/jira#686]
**Is your feature request related to a problem? Please describe.**
I'm writing telegram bot for interacting with Jira. The bot uses `aiogram` - asynchronous library, based on `aiohttp`. `jira` library uses blocking `requests` library.

**Describe the solution you'd like**
I propose to switch from `requests` library to `aiohttp` library for making requests. The only problem I see is that `aiohttp` supports only Python 3.5-3.7. That means that Python 2.7, 3.4 would be dropped.

**Describe alternatives you've considered**
For now, I will wrap all Jira related code to `asyncio.run_in_executor`.

**Comment by ssbarnea** [comment]
I can tell you from now that dropping py27 support is no go for the next two years at least and I would advocate against replacing requests with anything else too, as it became the defacto standard.

Async support is a very sensitive issue so I advise you to do an extensive research and comparison of possible solutions before writing any PR. 

Feel free to post all things you found here so we can make an informed decision regarding  director, mainly picking a safe-bet,one that would work long term.

**Comment by rominf** [comment]
@ssbarnea, thank you for your response.

**Comment by dchevell** [comment]
I’m not sure asyncio offers much value over threads in this library. The performance benefits of asyncio only become apparent when dealing with many hundreds or thousands of requests - if you benchmark aiottp vs using requests in a ThreadPoolExecutor with, say, 40 concurrent http requests, the time to completion is virtually identical. (FWIW I tested this before making this comment.)

In many use cases this is more than enough reason to prefer the asyncio approach, but not all use cases are equal. To reach the point where performance differences arise you’d need to be doing something like querying 100+ Jira endpoints simultaneously. Let’s put aside the unlikeliness of that - I’ll concede I can think of *some* use cases where I might need to do something with JQL that will match 5000+ issues (100 pages * 50 default per page). The real problem here is that those calls aren’t necessarily cheap - effectively forcing Jira to load 5k issues into memory at once. 

In other words, you’re likely start creating performance degradation in Jira before you ever reach a scale where asyncio would be worth it. You might even make things much worse, as you can’t avoid creating this load unless you introduce optional throttling mechanisms to prevent too many concurrent requests happening at once. At least with a ThreadPoolExecutor you can limit the maximum number of concurrent requests made to Jira by limiting the number of worker threads. 

All in all this would be a whole lot of work for a benefit that doesn’t really exist with our use case.
